File: languages/go/entity/identityV2/integer_enum_attribute_descriptor.go
package identityV2

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

// IntegerEnumAttributeDescriptor represents a descriptor for an attribute with an integer enum value.
type IntegerEnumAttributeDescriptor struct {
	AttributeDescriptor
	Enum []int64
}

// NewIntegerEnumAttributeDescriptor creates a new IntegerEnumAttributeDescriptor instance with the provided details.
func NewIntegerEnumAttributeDescriptor(name string, id string, description string, required bool, enum []int64) IntegerEnumAttributeDescriptor {
	a := AttributeDescriptor{
		DisplayName: name,
		Id:          id,
		Description: description,
		Required:    required,
	}
	return IntegerEnumAttributeDescriptor{
		AttributeDescriptor: a,
		Enum:                enum,
	}
}

func NewIntegerEnumAttributeDescriptorFromProto(s *proto.IntegerEnumAttributeDefinitionV2) IntegerEnumAttributeDescriptor {
	if s == nil {
		return IntegerEnumAttributeDescriptor{}
	}
	a := AttributeDescriptor{
		DisplayName: s.GetId(),
		Id:          s.Id,
		Description: s.Description,
		Required:    s.Required,
	}
	return IntegerEnumAttributeDescriptor{
		AttributeDescriptor: a,
		Enum:                s.GetEnum(),
	}
}

func (s IntegerEnumAttributeDescriptor) ToProto() *proto.IntegerEnumAttributeDefinitionV2 {
	return &proto.IntegerEnumAttributeDefinitionV2{
		DisplayName: s.DisplayName,
		Id:          s.Id,
		Description: s.Description,
		Required:    s.Required,
		Enum:        s.Enum,
	}
}

File: languages/python/bloock/entity/identity_v2/integer_enum_attribute_descriptor.py
from __future__ import annotations
from typing import List

import bloock._bridge.proto.identity_entities_v2_pb2 as proto
from bloock.entity.identity_v2.attribute_descriptor import AttributeDescriptor


class IntegerEnumAttributeDescriptor(AttributeDescriptor):

    def __init__(self, display_name: str, technical_name: str, description: str, required: bool, enumeration: List[int]) -> None:
        super().__init__(display_name, technical_name, description, required)
        self.enumeration = enumeration

    @staticmethod
    def from_proto(a: proto.IntegerEnumAttributeDefinitionV2) -> IntegerEnumAttributeDescriptor:
        return IntegerEnumAttributeDescriptor(
            display_name=a.display_name,
            technical_name=a.id,
            description=a.description,
            required=a.required,
            enumeration=list(a.enum),
        )

    def to_proto(self) -> proto.IntegerEnumAttributeDefinitionV2:
        return proto.IntegerEnumAttributeDefinitionV2(
            display_name=self.display_name,
            id=self.technical_name,
            description=self.description,
            required=self.required,
            enum=list(self.enumeration),
        )
